src/main/java/bio/terra/cli/apps/DockerAppsRunner.java

```diff
@@ -78,6 +86,24 @@ public boolean updateImageId(String imageId) {
     return true;
   }
 
+  // ====================================================
+  // Tool commands
+
+  /**
+   * Utility method for concatenating a command and its arguments.
+   *
+   * @param cmd command name (e.g. gsutil)
+   * @param cmdArgs command arguments (e.g. ls, gs://my-bucket)
+   */
+  public static String buildFullCommand(String cmd, List<String> cmdArgs) {
+    String fullCommand = cmd;
+    if (cmdArgs != null && cmdArgs.size() > 0) {
+      final String argSeparator = " ";
+      fullCommand += argSeparator + String.join(argSeparator, cmdArgs);
```

Comment:
very optional: using streams for this join `Stream.concat(Stream.of(cmd), cmdArgs.toStream()).collect(Collectors.joining(" "));`

Reply:
I prefer avoiding object creation unless it makes the code a lot clearer or faster somehow.